### ISSUE TITLE
Deterministic execution

### ISSUE THREAD

**Issue by lnicola** [IBM/terramind#43]
This is probably more of a TerraTorch question, but is it possible to make the execution more deterministic? Looking at `large_tile_generation.ipynb`, I tried adding:

```python
torch.manual_seed(42)
np.random.seed(42)
torch.backends.cudnn.deterministic = True
torch.backends.cudnn.benchmark = False
```

but even when running on `cpu`, the LULC output is very different. I didn't check the raw probabilities or logits, though.

<img width="322" height="505" alt="Image" src="https://github.com/user-attachments/assets/9e56c392-5a87-40fe-b4e8-823ad0ce4dc9" />

**Comment by blumenstiel** [comment]
It is fixed in the notebook. I assume your differences are coming from the dropout layers in the model.

**Comment by blumenstiel** [comment]
You are right, the seed should be correct. Then I also don't know where this difference could come from. @jhnnsjkbk1 Any idea why the generation is not deterministic with fixed seeds?

**Comment by lnicola** [comment]
TerraTorch uses the standard library's `random`, so one also needs to seed that.
